src/Microsoft.Health.Fhir.Shared.Api/Features/Resources/Bundle/BundleHandler.cs

```diff
@@ -242,17 +242,26 @@ public async Task<BundleResponse> Handle(BundleRequest request, CancellationToke
 
         private async Task PublishNotification(Hl7.Fhir.Model.Bundle responseBundle, BundleType bundleType)
         {
-            var apiCallResults = new Dictionary<string, int>();
+            var apiCallResults = new Dictionary<string, List<BundleSubCallMetricData>>();
             foreach (var entry in responseBundle.Entry)
             {
                 var status = entry.Response.Status;
                 if (apiCallResults.ContainsKey(status))
                 {
-                    apiCallResults[status]++;
+                    apiCallResults[status].Add(new BundleSubCallMetricData()
```

Comment:
Since the add is the same whether the key exists or not could it be moved out of the if statement to reduce duplication?

Reply:
Thanks. I cleaned up the duplicate code.